packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.Props.ts

```diff
@@ -354,9 +354,12 @@ export interface IContextualMenuItem {
    * Method to custom render this menu item.
    * For keyboard accessibility, the top-level rendered item should be a focusable element
    * (like an anchor or a button) or have the `data-is-focusable` property set to true.
+   *
+   * The function receives a function that can be called to dismiss the menu as a second argument.
+   *  This can be used to make sure that a custom menu item click dismisses the menu.
    * @defaultvalue undefined
    */
-  onRender?: (item: any) => React.ReactNode;
+  onRender?: (item: any, dismissMenu: (ev?: any, dismissAll?: boolean) => void) => React.ReactNode;
```

Comment:
Is there a reason that it's dismissMenu instead of onDismiss? Also, is there a reason we aren't passing in the entire IContextualMenuProps, in case we needed access to any of the other parent values? Maybe I'm not thinking about that function properly.

Lastly, is there a chance you can make the same changes to the commandbar in experiments? I hope to merge that in this week.

Reply:
So onDismiss to me communicates, "when the menu dismisses, call this function" where as dismissMenu means "call me if you want to dismiss the menu". 

So this isn't actually a prop that is being accessed from IContextualMenu, it's actually a private method.

I did make the same change to command bar in experiments as part of this PR.

Reply:
Or rather I made a change to make it compile, that command bar didn't have a dismiss function I could call

Reply:
yup, that makes sense. private function vs public callback. Just wondering if it belongs to an interface, and what happens when we need to pass another function/data down to the onRender.

Reply:
Which interface do you think it belongs on?

Reply:
dismiss is already a public function, would it make sense to have it on IContextualMenu, then pass that whole thing through? If you don't think it makes sense, i'm fine with it. Just curious.

Reply:
Hmm, I don't think we want to pass the whole context menu interface through just yet.

Reply:
I just meant IContextualMenu, not IContextualMenuProps. The former is currently empty.

Reply:
But it is a valid question of wether or not that interface makes sense to be passed through. Maybe IContextualMenuItem?

Reply:
I'd wait to do that until we have some meaningful stuff to put on that interface. I don't think dismiss is something that necessarily has to go on that interface either.

Reply:
mostly concerned with prop bloat. once we add that to onRender, it's public API and will be hard to clean up in the future. What happens when we need to pass more things into that onRender.

Reply:
So passing in IContextualMenuItem to onRender doesn't help solve the problem of menu dismissal, so I don't think it makes sense to introduce that as a part of this PR.

Reply:
yeah, i'm probably making mountains out of molehills. Works for me.

Reply:
No worries, I understand where you are coming from. Thanks for the review!